### PR TITLE
Feature/profile by link

### PR DESCRIPTION
**Pull request by Mykyta-Pysarev**
Remade profile page so it work with every user(not only authorised). You can access user profile simply clicking on username or name in tweets feed.
image 
image

Remade user reducer, so it has several redux stores (for auth and non-auth users). Therefore, a lot of changes in other components and containers, using user redux store.

### CODE REVIEW

File: src/components/FeedSingleTweet.tsx [hunk 1]
```diff
@@ -213,18 +222,21 @@ const FeedSingleTweet = ({
               <div>
                 <div
                   className=""
```
Comment (DavidKorystov) [inline src/components/FeedSingleTweet.tsx]:
```suggestion                  
```

File: src/components/FeedSingleTweet.tsx [hunk 2]
```diff
@@ -213,18 +222,21 @@ const FeedSingleTweet = ({
...
-                  dangerouslySetInnerHTML={{ __html: content }}
+                  dangerouslySetInnerHTML={{ __html: content || "" }}
                 />
                 <LinkPreview url={url[0]} />
               </div>
             ) : (
-              <div className="" dangerouslySetInnerHTML={{ __html: content }} />
+              <div
+                className=""
```
Comment (DavidKorystov) [inline src/components/FeedSingleTweet.tsx]:
```suggestion
```

File: src/containers/Banner.tsx [hunk 3]
```diff
@@ -76,25 +77,38 @@ const StyledInput = styled.input`
 
 interface BannerProps {
   isEditBanner: boolean;
```
Comment (DavidKorystov) [inline src/containers/Banner.tsx]: If you add an option, you don't have to write in isEditBanner={false} props, bcs undefined === false, right?
```suggestion
  isEditBanner?: boolean;
```

File: src/pages/Followers.tsx [hunk 4]
```diff
@@ -61,10 +61,10 @@ const FollowersPage: React.FunctionComponent = () => {
           </div>
           <div className="d-flex flex-column ms-2">
             <h1 className="fs-5 fw-bold ms-1">
-              {user.name ? user.name : "Full Name"}
+              {user.authUser.name ? user.authUser.name : "Full Name"}
```
Comment (DavidKorystov) [inline src/pages/Followers.tsx]: u can use i18 here

File: src/pages/Followers.tsx [hunk 5]
```diff
@@ -61,10 +61,10 @@ const FollowersPage: React.FunctionComponent = () => {
...
             </h1>
             <h3 className="fs-9 mt-1">
-              @{user.username ? user.username : "username"}
+              @{user.authUser.username ? user.authUser.username : "username"}
```
Comment (DavidKorystov) [inline src/pages/Followers.tsx]: u can use i18 here

File: src/pages/Following.tsx [hunk 6]
```diff
@@ -61,10 +61,10 @@ const FollowingPage: React.FunctionComponent = () => {
           </div>
           <div className="d-flex flex-column ms-2">
             <h1 className="fs-5 fw-bold ms-1">
-              {user.name ? user.name : "Full Name"}
+              {user.authUser.name ? user.authUser.name : "Full Name"}
```
Comment (DavidKorystov) [inline src/pages/Following.tsx]: u can use i18 here

File: src/pages/Following.tsx [hunk 7]
```diff
@@ -61,10 +61,10 @@ const FollowingPage: React.FunctionComponent = () => {
...
             </h1>
             <h3 className="fs-9 mt-1">
-              @{user.username ? user.username : "username"}
+              @{user.authUser.username ? user.authUser.username : "username"}
```
Comment (DavidKorystov) [inline src/pages/Following.tsx]: u can use i18 here

**Review by DavidKorystov — CHANGES_REQUESTED** [review]
Overall nice)

File: src/App.tsx [hunk 8]
```diff
@@ -52,13 +52,13 @@ const App: React.FunctionComponent = () => {
...
+        <Route path="/:id" element={<ProfilePage tabKey="tweets" />} />
         <Route
-          path="/profile/with_replies"
+          path="/:id/with_replies"
           element={<ProfilePage tabKey="tweets-with-replies" />}
         />
-        <Route path="/profile/media" element={<ProfilePage tabKey="media" />} />
-        <Route path="/profile/likes" element={<ProfilePage tabKey="likes" />} />
+        <Route path="/:id/media" element={<ProfilePage tabKey="media" />} />
+        <Route path="/:id/likes" element={<ProfilePage tabKey="likes" />} />
```
Comment (VitaliiDenysiuk) [inline src/App.tsx:61]: we have problem
1) clear site data
2) not login user
3) click any user
4)choose like
we have route with blank id. And white page.
image
image
image
image

**Review by VitaliiDenysiuk — CHANGES_REQUESTED** [review]

File: src/components/FeedSingleTweet.tsx (hunk 1 above)
Comment (Mykyta-Pysarev) [reply src/components/FeedSingleTweet.tsx]: fixed

File: src/components/FeedSingleTweet.tsx (hunk 2 above)
Comment (Mykyta-Pysarev) [reply src/components/FeedSingleTweet.tsx]: fixed

File: src/containers/Banner.tsx (hunk 3 above)
Comment (Mykyta-Pysarev) [reply src/containers/Banner.tsx]: True. Fixed.

File: src/pages/Followers.tsx (hunk 4 above)
Comment (Mykyta-Pysarev) [reply src/pages/Followers.tsx]: fixed

File: src/pages/Followers.tsx (hunk 5 above)
Comment (Mykyta-Pysarev) [reply src/pages/Followers.tsx]: fixed

File: src/pages/Following.tsx (hunk 6 above)
Comment (Mykyta-Pysarev) [reply src/pages/Following.tsx]: fixed

File: src/pages/Following.tsx (hunk 7 above)
Comment (Mykyta-Pysarev) [reply src/pages/Following.tsx]: fixed

File: src/App.tsx (hunk 8 above)
Comment (Mykyta-Pysarev) [reply src/App.tsx:61]: Question is why unauthorised user has access to tweets?

Comment (Mykyta-Pysarev) [reply src/App.tsx:61]: Fixed:
image

**Review by VitaliiDenysiuk — APPROVED** [review]
lgtm

File: src/components/FeedSingleTweet.tsx [hunk 9]
```diff
@@ -97,7 +97,7 @@ const FeedSingleTweet = ({
     "g"
   );
 
-  const url = content.match(urlSearch) || [];
+  const url = content ? content.match(urlSearch) || [] : [];
```
Comment (VitaliiDenysiuk) [inline src/components/FeedSingleTweet.tsx:100]: we have strange request:
image

**Review by VitaliiDenysiuk — CHANGES_REQUESTED** [review]

**Review by VitaliiDenysiuk — APPROVED** [review]
lgtm

File: src/components/FeedSingleTweet.tsx (hunk 9 above)
Comment (Mykyta-Pysarev) [reply src/components/FeedSingleTweet.tsx:100]: My bad. Fixed.

**Review by DavidKorystov — APPROVED** [review]
lgtm

**Pull request merged**